packages/core-api/src/routing/types.ts

```diff
@@ -27,6 +27,11 @@ export type RouteRef<Params extends { [param in string]: string } = {}> = {
 
 export type AnyRouteRef = RouteRef<any>;
 
+// @ts-ignore, we're just embedding the Params type for usage in other places
+export type ExternalRouteRef<
+  Params extends { [param in string]: string } = {}
+> = {};
```

Comment:
This type ends up being a bit weird, because it matches any object, including for example a `RouteRef`. So a function that accepts an `ExternalRouteRef` will accept most JS values 😅 . Tbh I think we probably need to go in the opposite direction and move `RouteRef` back to being a class

Reply:
Yeah, that is true. Everything is allowed here, I would love to narrow that down.